go/engine/track_token_test.go

```diff
@@ -76,3 +78,97 @@ func TestTrackTokenIdentify2(t *testing.T) {
 	defer runUntrack(tc.G, fu, username)
 	assertTracking(tc, username)
 }
+
+func TestTempTrackLocal(t *testing.T) {
+	tc := SetupEngineTest(t, "track")
+	defer tc.Cleanup()
+	fu := CreateAndSignupFakeUser(tc, "track")
+
+	fakeClock := clockwork.NewFakeClockAt(time.Now())
+	tc.G.Clock = fakeClock
+	// to pick up the new clock...
+	tc.G.ResetLoginState()
+
+	flakeyAPI := flakeyRooterAPI{orig: tc.G.XAPI, hardFail: false, G: tc.G}
+	tc.G.XAPI = &flakeyAPI
+
+	idUI := &FakeIdentifyUI{}
+	username := "t_tracy"
+
+	arg := &keybase1.Identify2Arg{
+		UserAssertion: username,
+		NeedProofSet:  true,
+	}
+	ctx := &Context{
+		LogUI:      tc.G.UI.GetLogUI(),
+		IdentifyUI: idUI,
+		SecretUI:   fu.NewSecretUI(),
+	}
+
+	// Identify tracy; all proofs should work
+	eng := NewResolveThenIdentify2(tc.G, arg)
+	if err := RunEngine(eng, ctx); err != nil {
+		t.Fatal(err)
+	}
+	targ := TrackTokenArg{
+		Token:   idUI.Token,
+		Options: keybase1.TrackOptions{BypassConfirm: true},
+	}
+
+	// Track tracy
+	teng := NewTrackToken(&targ, tc.G)
+	if err := RunEngine(teng, ctx); err != nil {
+		t.Fatal(err)
+	}
+
+	// Now make her Rooter proof fail
+	t.Logf("-- Set for Hard Fail --")
+	flakeyAPI.hardFail = true
```

Comment:
Why is this a hard failure? This mechanism (local caching/snoozing for temporary failures) really only ought to work for "soft" failures.